test/e2e/apimachinery/garbage_collector.go

```diff
@@ -1177,3 +1145,28 @@ var _ = SIGDescribe("Garbage collector", func() {
 		gatherMetrics(ctx, f)
 	})
 })
+
+// TODO(106575): Migrate away from generic polling function.
+func waitForReplicas(ctx context.Context, rc *v1.ReplicationController, rcClient clientv1.ReplicationControllerInterface) {
```

Comment:
by moving this into a helper function, do we lose visibility to which call to this helper failed from the main test body? if so, will that make debugging failures harder?

Reply:
I think there is a stacktrace because it is calling `Failf`, @pohly can you confirm?

Reply:
We should have a stacktrace for any failure in the helper.